src/UI/Components/Diagram/init.ts

```diff
@@ -161,11 +161,18 @@ export function diagramInit(
     },
 
     editEntity: (cellView, serviceModel, attributeValues) => {
+      let keyAttributes = serviceModel.key_attributes || [];
```

Comment:
Why do we have the exact same logic twice?

Reply:
extracted